### DISCUSSION TITLE
How to better organize (GET/POST/PUT) in NextJS API Routes?

### DISCUSSION THREAD

**Discussion by Gr8z** [vercel/next.js#11108] [Help]
This is how I am currently accounting for different types of API requests.

/pages/api/user.js
```js
export default (req, res) => {
  if (req.method === 'POST') {
    // Process a POST request
  } else {
    // Handle any other HTTP method
  }
}
```

It works fine but the file can get quite large after adding a bunch of stuff, most of the time unrelated.

Has anyone been working with a better organizational structure?

**Comment by timneutkens** [comment]
You can create pretty much any abstraction you want. For example:

Using a switch statement:

```js
export default (req, res) => {
  switch(req.method) {
    case 'POST': {
      res.json({message: 'This was a post'})
      break
    }
  }
}
```

Using a separate handlers object:

```js
const handlers = {
  post(req, res) {
    res.json({message: 'This was a post'})
  }
}

export default (req, res) => { 
  const execute = handlers[req.method.toLowerCase()]
  return execute(req, res)
}
```

**Reply by Arjan-Zuidema** [reply]
After 3 years we still have no better option than this?

Any thoughts on something like this: `<name>.get.ts` and `<name>.post.ts`?

**Reply by darwin403** [reply]
> After 3 years we still have no better option than this?
> 
> Any thoughts on something like this: `<name>.get.ts` and `<name>.post.ts`?

Next 13 has this covered. Check out the beta docs: https://beta.nextjs.org/docs/api-reference/file-conventions/route
